pkg/resources/management.cattle.io/v3/projectroletemplatebinding/validator.go

```diff
@@ -152,50 +142,65 @@ func clusterFromProject(project string) (string, string) {
 }
 
 // validUpdateFields checks if the fields being changed are valid update fields.
-func validateUpdateFields(oldPRTB, newPRTB *apisv3.ProjectRoleTemplateBinding) error {
-	var invalidFieldName string
+func validateUpdateFields(oldPRTB, newPRTB *apisv3.ProjectRoleTemplateBinding, fieldPath *field.Path) *field.Error {
+	const reason = "field is immutable"
```

Comment:
Due to the support matrix reasons, we can't use it now.
Separately I don't love putting actual logic inside code comments. The immutability check is simple and not harmful, but I fear that will cause more people to leverage [CEL](https://github.com/google/cel-spec) to write more involved logic in comments which is obviously more error-prone. So it seems like it is easier to just say no CEL for validation.

But I am curious if you have any counter thoughts.

Reply:
It would be nice to push as much validation as we can out of the webhook and into the API server. But with no way to write unit tests for code comments I agree with you that it's more error prone and probably not worth it.